### ISSUE TITLE
  Minify Resources (HTML, CSS, and JavaScript) to use less bandwidth (up to 28% less for html)

### ISSUE THREAD

**Issue by ghost** [searx/searx#942]
I was astonished when I discovered that searx.me didn't compress HTML CSS and JS. 

Using the HTML minifier recommended by Google Developers Guidelines on minifying resources: https://developers.google.com/speed/docs/insights/MinifyResources
one gets a 

> Original size: 11,013. Minified size: 7,842. Savings: 3,171 (28.79%).

for the front page of searx.me. This may save you quiet a bit of bandwidth with minimal effort.

**Comment by dalf** [comment]
I'm (slowly) working on it with a new theme : 
* see PR #858
* You can try here : https://al-f.net/searx/

Related to #687 #828

**Comment by lcpz** [comment]
@dalf Nice, just center again the results list and it'll be perfect, imho.

**Comment by dalf** [comment]
to make it similar to other search engines : 
![image](https://user-images.githubusercontent.com/1594191/29226060-e86bd57a-7ed0-11e7-97c2-07928af04b3a.png)
The screenshot is just a quick hack, it needs some works to make it really work.

I have the issue about the logo, should the one from oscar/pointhi be used or the one from oscar/logicodev ?

**Comment by asciimoo** [comment]
> I have the issue about the logo, should the one from oscar/pointhi be used or the one from oscar/logicodev ?

Well, this is a good question. There is no strict rule about which logo to use. The "official" logo is the magnifier with horns, but I think all the themes uses different visuals. So, feel free to use anything else if you want. =)

**Comment by lcpz** [comment]
I suggest to use the `searx` word in blue from main page, in a fashion similar to SP, Google and others.

**Comment by ghost** [comment]
Doesn't "minify" add a dependency on NodeJS? I would rather avoid this due to the impossible to build from source state of nodejs packages and their cyclic dependencies.

As long as searx can be build without nodejs as a dependency, it can be integrated into systems such as Guix (working on it). Systemintegration of Nodejs is generally difficult if you care about accountability.

**Comment by dalf** [comment]
I'm not sure to understand : 
https://github.com/asciimoo/searx/blob/master/searx/static/themes/simple/package.json
https://github.com/asciimoo/searx/blob/master/searx/static/themes/simple/gruntfile.js
https://github.com/asciimoo/searx/tree/master/searx/static/themes/simple/js

There are dependencies to grunt-cli and some npm packages. 
Same as the oscar theme.

**Comment by ghost** [comment]
Okay. then let me rephrase my last comment, written at a time of the night when you shouldn't be writting comments:

If you are just talking about modifications to the files which do not require this in the setup process
(so long as the build process does not demand, require npm packages) it's all good for me.

**Comment by dalf** [comment]
To deploy searx, there is no need to use npm.
To modify the oscar and simple themes, you need npm.

**Comment by dalf** [comment]
the simple theme PR was merged.

**Comment by lcpz** [comment]
But it's not centered?
